src/Containers/Microsoft.NET.Build.Containers/Tasks/CreateImageIndex.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Build.Framework;
+using Microsoft.Extensions.Logging;
+using Microsoft.NET.Build.Containers.Logging;
+using Microsoft.NET.Build.Containers.Resources;
+using ILogger = Microsoft.Extensions.Logging.ILogger;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.NET.Build.Containers.Tasks;
+
+public sealed partial class CreateImageIndex : Microsoft.Build.Utilities.Task, ICancelableTask, IDisposable
+{
+    private readonly CancellationTokenSource _cancellationTokenSource = new();
+
+    public void Cancel() => _cancellationTokenSource.Cancel();
+
+    public void Dispose()
+    {
+        _cancellationTokenSource.Dispose();
+    }
+
+    public override bool Execute()
+    {
+        try
+        {
+            Task.Run(() => ExecuteAsync(_cancellationTokenSource.Token)).GetAwaiter().GetResult();
+        }
+        catch (TaskCanceledException ex)
+        {
+            Log.LogWarningFromException(ex);
+        }
+        catch (OperationCanceledException ex)
+        {
+            Log.LogWarningFromException(ex);
+        }
+        return !Log.HasLoggedErrors;
+    }
+
+    internal async Task<bool> ExecuteAsync(CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+
+        var images = ParseImages();
+        if (Log.HasLoggedErrors)
+        {
+            return !Log.HasLoggedErrors;
+        }
+
+        using MSBuildLoggerProvider loggerProvider = new(Log);
+        ILoggerFactory msbuildLoggerFactory = new LoggerFactory(new[] { loggerProvider });
+        ILogger logger = msbuildLoggerFactory.CreateLogger<CreateImageIndex>();
+
+        logger.LogInformation(Strings.BuildingImageIndex, GetRepositoryAndTagsString(), string.Join(", ", images.Select(i => i.ManifestDigest)));
+
+        try
+        {
+            (string imageIndex, string mediaType) = ImageIndexGenerator.GenerateImageIndex(images);
+
+            GeneratedImageIndex = imageIndex;
```

Comment:
The fact that GeneratedImageIndex is declared in another file made me somewhat bamboozled since there is no other mention of it in this file as far as I could tell.
Might be just my unfamiliarity with the way Tasks are set up I suppose.
That being said, would
`(GeneratedImageIndex, string mediaType) = ...`
work in C#?

Reply:
Yeah, I placed the `CreateImageIndex` task parameters  in `CreateImageIndex.Interface.cs` similarly to `CreateNewImage` task. But now I se there are actually not a lot of parameters, so I will just move them to one file.

> That being said, would `(GeneratedImageIndex, string mediaType) = ...` work in C#?

yes, it will. Thanks, I'll apply the change

Reply:
done.